### PR TITLE
Improve api-check scripts to show more granular breaking changes.

### PR DESCRIPTION
**Pull request by pengdev**
### Summary of changes

This PR improve api-check scripts to show more granular breaking changes, and remove unnecessary prints.

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [x] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [ ] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog></changelog>`.
 - [ ] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` firstly and then port to `v10.[version]` release branch.

Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).

### CODE REVIEW

**Comment by pengdev** [comment]
A sample api-check CI output now looks like:

```
*** *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
*** *** *** ***         Potential major level breaking change found         *** *** *** ***
*** *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***

======================== locationcomponent ========================
Analysis results
----------------

Old API: tmp.tXWgX1twYw
New API: tmp.HVCL0ot45L
old: method void com.mapbox.maps.plugin.locationcomponent.LocationProviderImpl::<init>(android.content.Context)
new: method void com.mapbox.maps.plugin.locationcomponent.LocationProviderImpl::<init>(android.content.Context, com.mapbox.maps.plugin.locationcomponent.generated.LocationComponentSettings)
java.method.numberOfParametersChanged: The number of parameters of the method have changed.
SOURCE: BREAKING, BINARY: BREAKING

======================== sdk ========================
Analysis results
----------------

Old API: tmp.f7yKRuBevz
New API: tmp.7Bcvs5ap1U
old: method java.util.concurrent.atomic.AtomicBoolean com.mapbox.maps.renderer.MapboxRenderThread::getAwaitingNextVsync$sdk_release()
new: method boolean com.mapbox.maps.renderer.MapboxRenderThread::getAwaitingNextVsync$sdk_release()
java.method.returnTypeChanged: The return type changed from 'java.util.concurrent.atomic.AtomicBoolean' to  'boolean'.
SOURCE: POTENTIALLY_BREAKING, BINARY: BREAKING
java.annotation.removed: Element no longer annotated with 'org.jetbrains.annotations.NotNull'.
SOURCE: EQUIVALENT, SEMANTIC: POTENTIALLY_BREAKING, BINARY: EQUIVALENT

old: method boolean com.mapbox.maps.renderer.MapboxRenderThread::getEglPrepared$sdk_release()
new: <none>
java.method.removed: Method was removed.
SOURCE: BREAKING, BINARY: BREAKING

old: method java.util.concurrent.CopyOnWriteArrayList<java.lang.Runnable> com.mapbox.maps.renderer.MapboxRenderThread::getEventQueue$sdk_release()
new: <none>
java.method.removed: Method was removed.
SOURCE: BREAKING, BINARY: BREAKING

old: method java.util.concurrent.CopyOnWriteArrayList<java.lang.Runnable> com.mapbox.maps.renderer.MapboxRenderThread::getRenderEventQueue$sdk_release()
new: method java.util.concurrent.ConcurrentLinkedQueue<com.mapbox.maps.renderer.RenderEvent> com.mapbox.maps.renderer.MapboxRenderThread::getRenderEventQueue$sdk_release()
java.method.returnTypeChanged: The return type changed from 'java.util.concurrent.CopyOnWriteArrayList<java.lang.Runnable>' to  'java.util.concurrent.ConcurrentLinkedQueue<com.mapbox.maps.renderer.RenderEvent>'.
SOURCE: POTENTIALLY_BREAKING, BINARY: BREAKING

old: method java.util.concurrent.atomic.AtomicLong com.mapbox.maps.renderer.MapboxRenderThread::getRenderTimeNs$sdk_release()
new: method long com.mapbox.maps.renderer.MapboxRenderThread::getRenderTimeNs$sdk_release()
java.method.returnTypeChanged: The return type changed from 'java.util.concurrent.atomic.AtomicLong' to  'long'.
SOURCE: POTENTIALLY_BREAKING, BINARY: BREAKING
java.annotation.removed: Element no longer annotated with 'org.jetbrains.annotations.NotNull'.
SOURCE: EQUIVALENT, SEMANTIC: POTENTIALLY_BREAKING, BINARY: EQUIVALENT

old: method void com.mapbox.maps.renderer.MapboxRenderThread::queueEvent(java.lang.Runnable)
new: <none>
java.method.removed: Method was removed.
SOURCE: BREAKING, BINARY: BREAKING

old: parameter void com.mapbox.maps.renderer.MapboxRenderThread::queueRenderEvent(===java.lang.Runnable===)
new: parameter void com.mapbox.maps.renderer.MapboxRenderThread::queueRenderEvent(===com.mapbox.maps.renderer.RenderEvent===)
java.method.parameterTypeChanged: The type of the parameter changed from 'java.lang.Runnable' to 'com.mapbox.maps.renderer.RenderEvent'.
SOURCE: POTENTIALLY_BREAKING, BINARY: BREAKING

old: method void com.mapbox.maps.renderer.MapboxRenderThread::requestRender()
new: <none>
java.method.removed: Method was removed.
SOURCE: BREAKING, BINARY: BREAKING

old: method void com.mapbox.maps.renderer.MapboxRenderThread::setEglPrepared$sdk_release(boolean)
new: <none>
java.method.removed: Method was removed.
SOURCE: BREAKING, BINARY: BREAKING

old: method void com.mapbox.maps.renderer.MapboxRenderer::onDrawFrame()
new: <none>
java.method.removed: Method was removed.
SOURCE: BREAKING, BINARY: BREAKING

old: method void com.mapbox.maps.renderer.MapboxRenderer::onSurfaceCreated()
new: <none>
java.method.removed: Method was removed.
SOURCE: BREAKING, BINARY: BREAKING

old: method void com.mapbox.maps.renderer.MapboxRenderer::onSurfaceDestroyed()
new: <none>
java.method.removed: Method was removed.
SOURCE: BREAKING, BINARY: BREAKING

old: method void com.mapbox.maps.renderer.RenderHandlerThread::clearMessageQueue()
new: method void com.mapbox.maps.renderer.RenderHandlerThread::clearMessageQueue(boolean)
java.method.numberOfParametersChanged: The number of parameters of the method have changed.
SOURCE: BREAKING, BINARY: BREAKING

old: method boolean com.mapbox.maps.renderer.egl.EGLCore::getEglStatusSuccess()
new: <none>
java.method.removed: Method was removed.
SOURCE: BREAKING, BINARY: BREAKING

old: method void com.mapbox.maps.renderer.egl.EGLCore::makeNothingCurrent()
new: method boolean com.mapbox.maps.renderer.egl.EGLCore::makeNothingCurrent()
java.method.returnTypeChanged: The return type changed from 'void' to  'boolean'.
SOURCE: POTENTIALLY_BREAKING, BINARY: BREAKING

old: method void com.mapbox.maps.renderer.egl.EGLCore::prepareEgl()
new: method boolean com.mapbox.maps.renderer.egl.EGLCore::prepareEgl()
java.method.returnTypeChanged: The return type changed from 'void' to  'boolean'.
SOURCE: POTENTIALLY_BREAKING, BINARY: BREAKING

old: method void com.mapbox.maps.renderer.egl.EGLCore::setEglStatusSuccess(boolean)
new: <none>
java.method.removed: Method was removed.
SOURCE: BREAKING, BINARY: BREAKING

old: method void com.mapbox.maps.viewannotation.ViewAnnotation::<init>(android.view.View, boolean, boolean, android.widget.FrameLayout$LayoutParams)
new: method void com.mapbox.maps.viewannotation.ViewAnnotation::<init>(android.view.View, android.view.View$OnAttachStateChangeListener, boolean, boolean, android.widget.FrameLayout$LayoutParams, int, int)
java.method.numberOfParametersChanged: The number of parameters of the method have changed.
SOURCE: BREAKING, BINARY: BREAKING

old: method boolean com.mapbox.maps.viewannotation.ViewAnnotation::component2()
new: method android.view.View$OnAttachStateChangeListener com.mapbox.maps.viewannotation.ViewAnnotation::component2()
java.method.returnTypeChanged: The return type changed from 'boolean' to  'android.view.View$OnAttachStateChangeListener'.
SOURCE: POTENTIALLY_BREAKING, BINARY: BREAKING
java.annotation.added: Element newly annotated with 'org.jetbrains.annotations.Nullable'.
SOURCE: EQUIVALENT, SEMANTIC: POTENTIALLY_BREAKING, BINARY: EQUIVALENT

old: method android.widget.FrameLayout$LayoutParams com.mapbox.maps.viewannotation.ViewAnnotation::component4()
new: method boolean com.mapbox.maps.viewannotation.ViewAnnotation::component4()
java.method.returnTypeChanged: The return type changed from 'android.widget.FrameLayout$LayoutParams' to  'boolean'.
SOURCE: POTENTIALLY_BREAKING, BINARY: BREAKING
java.annotation.removed: Element no longer annotated with 'org.jetbrains.annotations.NotNull'.
SOURCE: EQUIVALENT, SEMANTIC: POTENTIALLY_BREAKING, BINARY: EQUIVALENT

old: method com.mapbox.maps.viewannotation.ViewAnnotation com.mapbox.maps.viewannotation.ViewAnnotation::copy(android.view.View, boolean, boolean, android.widget.FrameLayout$LayoutParams)
new: method com.mapbox.maps.viewannotation.ViewAnnotation com.mapbox.maps.viewannotation.ViewAnnotation::copy(android.view.View, android.view.View$OnAttachStateChangeListener, boolean, boolean, android.widget.FrameLayout$LayoutParams, int, int)
java.method.numberOfParametersChanged: The number of parameters of the method have changed.
SOURCE: BREAKING, BINARY: BREAKING

old: method android.view.ViewTreeObserver$OnGlobalLayoutListener com.mapbox.maps.viewannotation.ViewAnnotation::getGlobalLayoutListener()
new: <none>
java.method.removed: Method was removed.
SOURCE: BREAKING, BINARY: BREAKING

old: method void com.mapbox.maps.viewannotation.ViewAnnotation::setGlobalLayoutListener(android.view.ViewTreeObserver$OnGlobalLayoutListener)
new: <none>
java.method.removed: Method was removed.
SOURCE: BREAKING, BINARY: BREAKING

======================== sdk-base ========================
Analysis results
----------------

Old API: tmp.4nyE8CUFXa
New API: tmp.nX0R6MsJZB
old: method com.mapbox.maps.plugin.LocationPuck com.mapbox.maps.plugin.locationcomponent.generated.LocationComponentSettings::component7()
new: method com.mapbox.maps.plugin.PuckBearingSource com.mapbox.maps.plugin.locationcomponent.generated.LocationComponentSettings::component7()
java.method.returnTypeChanged: The return type changed from 'com.mapbox.maps.plugin.LocationPuck' to  'com.mapbox.maps.plugin.PuckBearingSource'.
BINARY: BREAKING, SOURCE: POTENTIALLY_BREAKING

old: method com.mapbox.maps.plugin.locationcomponent.generated.LocationComponentSettings com.mapbox.maps.plugin.locationcomponent.generated.LocationComponentSettings::copy(boolean, boolean, int, float, java.lang.String, java.lang.String, com.mapbox.maps.plugin.LocationPuck)
new: method com.mapbox.maps.plugin.locationcomponent.generated.LocationComponentSettings com.mapbox.maps.plugin.locationcomponent.generated.LocationComponentSettings::copy(boolean, boolean, int, float, java.lang.String, java.lang.String, com.mapbox.maps.plugin.PuckBearingSource, com.mapbox.maps.plugin.LocationPuck)
java.method.numberOfParametersChanged: The number of parameters of the method have changed.
BINARY: BREAKING, SOURCE: BREAKING

old: <none>
new: method com.mapbox.maps.plugin.PuckBearingSource com.mapbox.maps.plugin.locationcomponent.generated.LocationComponentSettingsInterface::getPuckBearingSource()
java.method.addedToInterface: Method was added to an interface.
SEMANTIC: POTENTIALLY_BREAKING, BINARY: NON_BREAKING, SOURCE: BREAKING

old: <none>
new: method void com.mapbox.maps.plugin.locationcomponent.generated.LocationComponentSettingsInterface::setPuckBearingSource(com.mapbox.maps.plugin.PuckBearingSource)
java.method.addedToInterface: Method was added to an interface.
SEMANTIC: POTENTIALLY_BREAKING, BINARY: NON_BREAKING, SOURCE: BREAKING
```

**Comment by pengdev** [comment]
The internal classes can not be easily ignored as the revapi doesn't understand Kotlin metadata.  Will investigate other tools that can checks api compatibility for Kotlin.

**Review by tobrun — APPROVED** [review]

File: scripts/java-api-check-all.sh [hunk 1]
```diff
@@ -1,11 +1,12 @@
 #!/usr/bin/env bash
 
-set -Eeuxo pipefail
+set -Eeuo pipefail
 # Usage:
 #   ./java-api-check-all.sh <current release tag (empty for branches)>
```
Comment (ank27) [inline scripts/java-api-check-all.sh:5]: Can we add a note to build the project before running `java-api-check-all.sh` while running locally?

**Review by ank27 — APPROVED** [review]

**Comment by kiryldz** [comment]
> As noted above, many breaking changes are related to the Kotlin internal class, which shouldn't be treated as public API breaking change.
> 
> The reason is `revapi` we used under the hood doesn't understand Kotlin metadata. Will investigate other tools that can checks api compatibility for Kotlin.

Technically Kotlin `internal` class are indeed Java public ones. 
@pengdev I did not test it myself but can [`@JvmSynthetic`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.jvm/-jvm-synthetic/) help somehow here?

**Review by kiryldz — APPROVED** [review]

**Comment by pengdev** [comment]
> Technically Kotlin internal class are indeed Java public ones.
> @pengdev I did not test it myself but can [@JvmSynthetic](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.jvm/-jvm-synthetic/) help somehow here?

I think we need to have a separate bot to check the Kotlin API compatibility in parallel with the Java API checker. 

And introducing [@JvmSynthetic](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.jvm/-jvm-synthetic/) annotation will also cause breaking changes to our Java APIs, but I think that would reduce further Java API breaking changes, I think we could optimise our internal classes to use the `JvmSynthetic` annotation.

**Pull request merged**